lighthouse-core/fraggle-rock/gather/driver.js

```diff
@@ -58,6 +67,7 @@ class Driver {
     const session = await this._page.target().createCDPSession();
     this._session = this.defaultSession = new ProtocolSession(session);
     this._executionContext = new ExecutionContext(this._session);
+    this._fetcher = new Fetcher(this);
```

Comment:
I've avoided passing `this` so far, WDYT about passing `session` and `executionContext` instead?